src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -2574,40 +2582,42 @@ static int stop_xmc_nolock(struct platform_device *pdev)
 			READ_REG32(xmc, XMC_VERSION_REG),
 			READ_REG32(xmc, XMC_STATUS_REG), magic);
 
-		reg_val = READ_REG32(xmc, XMC_STATUS_REG);
-		if (!(reg_val & STATUS_MASK_STOPPED)) {
-			xocl_info(&xmc->pdev->dev, "Stopping XMC...");
-			WRITE_REG32(xmc, CTL_MASK_STOP, XMC_CONTROL_REG);
-			WRITE_REG32(xmc, 1, XMC_STOP_CONFIRM_REG);
-		}
-
-		retry = 0;
-		while (retry++ < MAX_XMC_RETRY &&
-			!(READ_REG32(xmc, XMC_STATUS_REG) &
-			STATUS_MASK_STOPPED))
-			msleep(RETRY_INTERVAL);
+		u32 misc = READ_VERSION_CTRL_REG32(xmc, VERSION_CTRL_MISC_REG);
```

Comment:
This check applies to all platforms, correct? Will this register be zero for other platforms? (u200, u250, etc)

Reply:
Not guaranteed actually, thanks for pointing this case. I will add one more check as this change is needed only for SmartSSD U.2

Reply:
> Not guaranteed actually, thanks for pointing this case. I will add one more check as this change is needed only for SmartSSD U.2
Maybe you may define "__RES_VERSION_CTRL" only for U2 in devices.h?